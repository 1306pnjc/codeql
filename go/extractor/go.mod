module github.com/github/codeql-go/extractor

go 1.22.0

// when updating this, run
//    bazel run @rules_go//go -- mod tidy
// when adding or removing dependencies, run
//    bazel mod tidy
require (
	golang.org/x/mod v0.16.0
	golang.org/x/tools v0.18.0
)
