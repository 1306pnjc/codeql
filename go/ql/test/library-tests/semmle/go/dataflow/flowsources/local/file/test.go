package test

import "os"

func open() {
	file, err := os.Open("file.txt")
	if err != nil {
		return
	}
	defer file.Close()
	file.Read([]byte{1, 2, 3})
}

func openFile() {
	file, err := os.OpenFile("file.txt", os.O_RDWR, 0)
	if err != nil {
		return
	}
	defer file.Close()
	file.Read([]byte{1, 2, 3})
}

func readFile() {
	data, err := os.ReadFile("file.txt")
	if err != nil {
		return
	}
	_ = data
}
