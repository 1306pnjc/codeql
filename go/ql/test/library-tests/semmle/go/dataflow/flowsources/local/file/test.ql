import go

from DataFlow::Node source
where source instanceof ThreatModelFlowSource
select source
