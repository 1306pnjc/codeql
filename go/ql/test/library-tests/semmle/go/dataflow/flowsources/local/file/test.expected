| test.go:6:2:6:33 | ... := ...[0] |
| test.go:15:2:15:51 | ... := ...[0] |
| test.go:24:2:24:37 | ... := ...[0] |
